### PR TITLE
fix: valid type "member" in "interactionCreate" event

### PR DESCRIPTION
**Pull request by ayntee**

### CODE REVIEW

File: src/api/controllers/interactions.ts [hunk 1]
```diff
@@ -4,17 +4,13 @@ import {
...
 
-export async function handleInternalInteractionCreate(data: DiscordPayload) {
+export function handleInternalInteractionCreate(data: DiscordPayload) {
   if (data.t !== "INTERACTION_CREATE") return;
 
   const payload = data.d as InteractionCommandPayload;
   eventHandlers.interactionCreate?.(
-    {
-      ...payload,
-      member: await structures.createMember(payload.member, payload.guild_id),
```
Comment (itohatweb) [inline src/api/controllers/interactions.ts:16]: Why remove member from event?

File: src/api/structures/guild.ts [hunk 2]
```diff
@@ -210,7 +210,7 @@ export async function createGuild(data: CreateGuildPayload, shardID: number) {
 
   initialMemberLoadQueue.set(guild.id, members);
 
-  return guild;
+  return guild as Guild;
```
Comment (itohatweb) [inline src/api/structures/guild.ts:213]: Irrelevant change, should be done in a other PR.

File: src/api/structures/member.ts [hunk 3]
```diff
@@ -126,7 +126,7 @@ export async function createMember(data: MemberCreatePayload, guildID: string) {
 
   await cacheHandlers.set("members", member.id, member);
 
-  return member;
+  return member as Member;
```
Comment (itohatweb) [inline src/api/structures/member.ts:129]: Same as above

File: src/api/structures/template.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ export function createTemplate(
     sourceGuildID: createNewProp(sourceGuildID),
     serializedSourceGuild: createNewProp(serializedSourceGuild),
     isDirty: createNewProp(isDirty),
-  });
+  }) as Template;
```
Comment (itohatweb) [inline src/api/structures/template.ts:43]: Same as above

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: src/api/controllers/interactions.ts (hunk 1 above)
Comment (ayntee) [reply src/api/controllers/interactions.ts:16]: `member` still exists in the data, but only the type and value is altered.

Comment (itohatweb) [reply src/api/controllers/interactions.ts:16]: Well yes member still exists but member does not get cached into memory. Also all other events either create the member or get it from cache. So why removing it here?

Comment (ayntee) [reply src/api/controllers/interactions.ts:16]: The type of the first parameter of the "interactionCreate" event is `InteractionCommandPayload`, it includes the "member" property, which is of type `MemberCreatePayload`, **not** `Member`; however, the `createMember()` function resolves to the type `Member`, consequently conflict arises, to resolve this conflict, the raw member payload is passed to the "interactionCreate" event.

Comment (itohatweb) [reply src/api/controllers/interactions.ts:16]: Then interaction create event should return the interaction and the member imo.

Comment (ayntee) [reply src/api/controllers/interactions.ts:16]: Currently, the "interactionCreate" event has one parameter *data* which is of type `InteractionCommandPayload`, however, if I were to remove this, the type "Member" and "MemberCreatePayload" would result in errors.

Comment (itohatweb) [reply src/api/controllers/interactions.ts:16]: What I mean is not remove it but add a second arg for the member.

**Pull request merged**